frontend/src/actions/WindowActions.js

```diff
@@ -521,38 +528,64 @@ export function deselectTableItems(ids, windowType, viewId) {
 
 // THUNK ACTIONS
 
-export function initTabs(layout, windowType, docId, isModal) {
-  return async (dispatch) => {
-    const requests = [];
-    const tabTmp = {};
-
-    if (layout) {
-      layout.map((tab, index) => {
-        tabTmp[tab.tabId] = {};
-
-        if ((tab.tabId && index === 0) || !tab.queryOnActivate) {
-          requests.push(getTab(tab.tabId, windowType, docId));
-        }
-      });
+// TODO: Just a quick thunk action creator to test Tables reducer
+// but looks like this can actually replace the `initTabs`.
+/*
+ * @method fetchTab
+ * @summary Action creator for fetching single tab's rows
+ */
+export function fetchTab(tabId, windowType, docId) {
+  return (dispatch) => {
+    return getTabRequest(tabId, windowType, docId)
+      .then((response) => {
+        const tableId = `${windowType}_${docId}_${tabId}`;
+        const tableData = { result: response };
 
-      return await Promise.all(requests).then((responses) => {
-        responses.forEach((res) => {
-          // needed for finding tabId
-          const rowZero = res && res[0];
-          if (rowZero) {
-            const tabId = rowZero.tabId;
-            tabTmp[tabId] = res;
-          }
-        });
+        dispatch(updateTabTable(tableId, tableData));
 
-        dispatch(addRowData(tabTmp, getScope(isModal)));
+        return Promise.resolve(response);
+      })
+      .catch((error) => {
+        //show error message ?
+        return Promise.resolve(error);
       });
-    }
-
-    return Promise.resolve(null);
   };
 }
 
+// TODO: Figure out if we still need this as it looks like we can just fetch
```

Comment:
Yeeah so I'm kinda reluctant to do it at this point as it's gonna be easier to revert it by just uncommenting code imho.

Reply:
Maybe leave some small comment in here instead with link to the PR/commit in which we removed the function to find our way around quick in case of an urgent fix/revert ?